lib/kafka/connection.rb

```diff
@@ -88,10 +93,10 @@ def send_request(request)
 
         @correlation_id += 1
 
-        write_request(request, notification)
+        @operation.write_request(request, notification, @correlation_id)
 
         response_class = request.response_class
-        wait_for_response(response_class, notification) unless response_class.nil?
+        @operation.wait_for_response(response_class, notification, @correlation_id) unless response_class.nil?
```

Comment:
I don't see why we need to wrap Connection here – why not just instantiate the handshake connection and execute it here? Why does `write_request` and `wait_for_response` have to move?